Tests/Adapter/SymfonyCacheTest.php

```diff
@@ -111,4 +111,146 @@ public function testCacheActionWithInvalidType()
+            $this->router,
+            $this->filesystem,
+            '/cache/dir',
+            'token',
+            false,
+            array('all', 'translations'),
+            array(
+                array('ip' => '213.186.35.9', 'domain' => 'www.example.com', 'basic' => false, 'port' => 80),
+            )
+        );
+
+        $that = $this;
+        $mocks = array();
+
+        $builder = new $mockBuilderClass();
+        $mock = $builder->setNamespace('Sonata\CacheBundle\Adapter')
+            ->setName('socket_create')
+            ->setFunction(function () use ($that) {
+                $that->assertSame(array(AF_INET, SOCK_STREAM, SOL_TCP), func_get_args());
+            })
+            ->build();
+        $mock->enable();
+
+        $mocks[] = $mock;
+
+        foreach (array('socket_set_option', 'socket_connect', 'socket_write', 'socket_read') as $function) {
+            $builder = new $mockBuilderClass();
+            $mock = $builder->setNamespace('Sonata\CacheBundle\Adapter')
+                ->setName($function)
+                ->setFunction(function () {
+                })
+                ->build();
+            $mock->enable();
+
+            $mocks[] = $mock;
+        }
+
+        $cache->flush();
+
+        foreach ($mocks as $mock) {
+            $mock->disable();
+        }
+    }
+
+    /**
+     * Tests the flush method with IPv6.
+     */
+    public function testFlushWithIPv6()
+    {
+        $mockBuilderClass = $this->getMockBuilderClass();
+
+        $cache = new SymfonyCache(
+            $this->router,
+            $this->filesystem,
+            '/cache/dir',
+            'token',
+            false,
+            array('all', 'translations'),
+            array(
+                array('ip' => '2001:41d0:1:209:FF:FF:FF:FF', 'domain' => 'www.example.com', 'basic' => false, 'port' => 80),
+            )
+        );
+
+        $that = $this;
+        $mocks = array();
+
+        $builder = new $mockBuilderClass();
+        $mock = $builder->setNamespace('Sonata\CacheBundle\Adapter')
+            ->setName('socket_create')
+            ->setFunction(function () use ($that) {
+                $that->assertSame(array(AF_INET6, SOCK_STREAM, SOL_TCP), func_get_args());
+            })
+            ->build();
+        $mock->enable();
+
+        $mocks[] = $mock;
+
+        foreach (array('socket_set_option', 'socket_connect', 'socket_write', 'socket_read') as $function) {
+            $builder = new $mockBuilderClass();
+            $mock = $builder->setNamespace('Sonata\CacheBundle\Adapter')
+                ->setName($function)
+                ->setFunction(function () {
+                })
+                ->build();
+            $mock->enable();
+
+            $mocks[] = $mock;
+        }
+
+        $cache->flush();
+
+        foreach ($mocks as $mock) {
+            $mock->disable();
+        }
+    }
+
+    /**
+     * Gets the mock builder class according to the lib version (ie the PHP version).
```

Comment:
In this comment, please add `NEXT_MAJOR : while dropping old versions of php, restrict the library to 1.0 and simplify this`.